### ISSUE TITLE
Bootstrap error copying Terraform state to Cloud Storage

### ISSUE THREAD

**Issue by jtblin** [terraform-google-modules/terraform-example-foundation#977]
### TL;DR

In the step "Re-run terraform init. When you're prompted, agree to copy Terraform state to Cloud Storage."

I get the following error:

```
Initializing the backend...
Initializing modules...
╷
│ Error: Error inspecting states in the "local" backend:
│     querying Cloud Storage failed: storage: bucket doesn't exist```

Note that in the previous step (10), there was an error and there was no "UPDATE_PROJECTS_BACKEND" placement.

I do not have any project created yet apparently so not sure where it is supposed to create the bucket to save the state.


### Expected behavior

State is saved to the bucket


### Observed behavior

```
Initializing the backend...
Initializing modules...
╷
│ Error: Error inspecting states in the "local" backend:
│     querying Cloud Storage failed: storage: bucket doesn't exist```

### Terraform Configuration

```hcl
My backend.tf looks like


terraform {
  backend "gcs" {
    bucket = "bkt-prj-b-seed-tfstate-7030"
    prefix = "terraform/bootstrap/state"
  }
}
```
```


### Terraform Version

```sh
Terraform v1.5.0
on darwin_arm64
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/google v4.69.1
+ provider registry.terraform.io/hashicorp/google-beta v4.69.1
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

**Comment by jtblin** [comment]
Looks like it was a default project issue with gcloud or eventual consistency issue or something like that. Changing default project fixed the issue and I was able to save the state to the bucket.

**Comment by Ariestattoo** [comment]
@jtblin What did you do exactly to fix this? I've set the project to the seed and cicd project and I still get this error. The bucket is there and I have access...but still 
 Error: Error inspecting states in the "local" backend:
│     querying Cloud Storage failed: storage: bucket doesn't exist

**Comment by jtblin** [comment]
Yes that `gcloud auth application-default login --project $project_id`, I think I was just not logged in the proper project.
